go/home/home.go

```diff
@@ -370,3 +386,56 @@ func (h *Home) Create(ctx context.Context, cli gregor1.IncomingInterface, catego
 func (h *Home) Dismiss(ctx context.Context, cli gregor1.IncomingInterface, category string, ibm gregor.Item) (bool, error) {
 	return true, nil
 }
+
+type rawPollHome struct {
+	Status       libkb.AppStatus `json:"status"`
+	NextPollSecs int             `json:"next_poll_secs"`
+}
+
+func (r *rawPollHome) GetAppStatus() *libkb.AppStatus {
+	return &r.Status
+}
+
+func (h *Home) RunUpdateLoop(m libkb.MetaContext) {
+	go h.updateLoopThread(m)
+}
+
+func (h *Home) updateLoopThread(m libkb.MetaContext) {
+	m = m.WithLogTag("HULT")
+	m.CDebugf("Starting Home#updateLoopThread")
+	slp := time.Minute * (time.Duration(5) + time.Duration((rand.Int() % 10)))
+	var err error
+	for {
+		m.CDebugf("Sleeping %v until next poll", slp)
+		m.G().Clock().Sleep(slp)
+		slp, err = h.pollOnce(m)
+		if _, ok := err.(libkb.DeviceRequiredError); ok {
+			slp = time.Duration(1) * time.Minute
+		} else if err != nil {
+			slp = time.Duration(15) * time.Minute
+			m.CDebugf("Hit an error in home update loop: %v", err)
+		}
+	}
+}
+
+func (h *Home) pollOnce(m libkb.MetaContext) (d time.Duration, err error) {
+	defer m.CTraceTimed("Home#pollOnce", func() error { return err })()
+
+	if !m.HasAnySession() {
+		m.CDebugf("No-op, since don't have keys (and/or am not logged in)")
+		return time.Duration(0), libkb.DeviceRequiredError{}
+	}
+
+	var raw rawPollHome
+	err = m.G().API.GetDecode(libkb.APIArg{
+		Endpoint:    "home/poll",
+		SessionType: libkb.APISessionTypeREQUIRED,
+		Args:        libkb.HTTPArgs{},
+		MetaContext: m,
+	}, &raw)
```

Comment:
to clarify: this polls the server, which will send down a gregor message consumed on another path?

Reply:
yes, that is the flow. It forces the server to refresh its state, so that new home state, and new badging will be recomputed.